src/vault/vault.ts

```diff
@@ -121,47 +153,102 @@ export class Vault {
     context: KeyContext,
     associatedData?: string,
   ): Promise<string> {
-    const { dataKey, encryptedKeys } = await this.createDataKey({
+    const keyPair = await this.createDataKey({
       context,
     });
-    return encrypt(data, dataKey.key, encryptedKeys, associatedData || '');
+
+    // Convert base64 key to Uint8Array
+    const encoder = new TextEncoder();
+
+    // Decode the base64 key string to a binary string, then to a Uint8Array
+    const keyBase64Decoded = atob(keyPair.dataKey.key);
+    const key = new Uint8Array(keyBase64Decoded.length);
+    for (let i = 0; i < keyBase64Decoded.length; i++) {
+      key[i] = keyBase64Decoded.charCodeAt(i);
+    }
+
+    // Same for encrypted keys
+    const encKeysBase64Decoded = atob(keyPair.encryptedKeys);
+    const keyBlob = new Uint8Array(encKeysBase64Decoded.length);
+    for (let i = 0; i < encKeysBase64Decoded.length; i++) {
+      keyBlob[i] = encKeysBase64Decoded.charCodeAt(i);
+    }
+
+    const prefixLenBuffer = encodeUInt32(keyBlob.length);
+    const aadBuffer = associatedData
+      ? encoder.encode(associatedData)
+      : undefined;
+
+    // Use a 12-byte IV for AES-GCM (industry standard)
+    const iv = this.cryptoProvider.randomBytes(12);
+
+    const {
+      ciphertext,
+      iv: resultIv,
+      tag,
+    } = await this.cryptoProvider.encrypt(
+      encoder.encode(data),
+      key,
+      iv,
+      aadBuffer,
+    );
+
+    // Concatenate all parts into a single array
+    const resultArray = new Uint8Array(
+      resultIv.length +
+        tag.length +
+        prefixLenBuffer.length +
+        keyBlob.length +
+        ciphertext.length,
+    );
+
+    let offset = 0;
+    resultArray.set(resultIv, offset);
+    offset += resultIv.length;
+
+    resultArray.set(tag, offset);
+    offset += tag.length;
+
+    resultArray.set(new Uint8Array(prefixLenBuffer), offset);
+    offset += prefixLenBuffer.length;
+
+    resultArray.set(keyBlob, offset);
+    offset += keyBlob.length;
+
+    resultArray.set(ciphertext, offset);
+
+    // Convert to base64
+    const resultBuffer = Buffer.from(resultArray);
+    return resultBuffer.toString('base64');
   }
 
   async decrypt(
     encryptedData: string,
     associatedData?: string,
   ): Promise<string> {
-    const decoded = decode(encryptedData);
+    const decoded = this.decode(encryptedData);
     const dataKey = await this.decryptDataKey({ keys: decoded.keys });
-    return decrypt(decoded, dataKey.key, associatedData || '');
-  }
 
-  /*
-   * @deprecated Use `createObject` instead.
-   */
-  createSecret = this.createObject;
```

Comment:
Is this going out as a new major version? The collection of `*Secrets` methods were released in the first nodejs Vault integration, removing them fully will be a breaking change.